### PR TITLE
PR author's checklist

### PR DESCRIPTION
**Pull request by gcongiu**
Add PR author's checklist for github

### CODE REVIEW

**Comment by gcongiu** [comment]
@jagode @adanalis @anustuvicl @dbarry9 anything you would like to add the the checklist?

**Comment by dbarry9** [comment]
The list looks good to me.

`The PR needs to pass all the tests.`

I know that some tests can fail if they take too long (e.g. > 60 minutes). This happened to one of my PR's with the ROCM test. I am not sure the cause of this, but this is a reason a test might not pass, through no fault of the PR author.

**Comment by gcongiu** [comment]
@dbarry9 generally, all the tests should pass. If some don't we can evaluate whether to merge anyway or not on a case basis (like you said).

**Comment by dbarry9** [comment]
I agree.

**Comment by gcongiu** [comment]
@G-Ragghianti we might not want to run tests for PRs like this one. Is there a way of triggering tests on demand with GitHub actions, e.g., by adding a comment to the PR?

**Comment by G-Ragghianti** [comment]
There is a way, but I would prefer to run a few occasional tests
unnecessarily rather than turn off automated testing.

**Comment by gcongiu** [comment]
@G-Ragghianti ok let's keep it this way. We can always change the default in the future if the need arises.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Pull Request Description
+
+
+## Author Checklist
+- [ ] **Description**
```
Comment (jagode) [inline .github/pull_request_template.md:5]: Checklist for Pull Request Authors

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+_Why_ this PR exists. Reference all relevant information including _background_, _issues_, _test failures_, etc
```
Comment (jagode) [inline .github/pull_request_template.md:6]: Include _Why_ this PR exists. 
(Comma after 'information')

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+- [ ] **Commits**
+_Commits_ are self contained and only do one thing
+_Commits_ have header of the form: `module: short description`
```
Comment (jagode) [inline .github/pull_request_template.md:9]: a header

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+_Commits_ have body (whenever relevant) containing detailed description of the addressed problem and its solution
```
Comment (jagode) [inline .github/pull_request_template.md:10]: a body...
a detailed...

**Review by jagode — CHANGES_REQUESTED** [review]
I would make it a little bit more descriptive to enhance readability. See suggestions inline.

File: .github/pull_request_template.md (hunk 3 above)
Comment (gcongiu) [reply .github/pull_request_template.md]: done

File: .github/pull_request_template.md (hunk 4 above)
Comment (gcongiu) [reply .github/pull_request_template.md]: done

**Comment by gcongiu** [comment]
@jagode I update the PR

**Review by jagode — APPROVED** [review]
Looks good now.

**Pull request merged**